samples/dect/dect_phy/dect_shell/CMakeLists.txt

```diff
@@ -0,0 +1,47 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+cmake_minimum_required(VERSION 3.20.0)
+
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(DESH)
+target_include_directories(app PRIVATE src)
+
+target_sources(app PRIVATE src/main.c)
+target_sources(app PRIVATE src/shell.c)
+
+add_subdirectory(src/print)
+add_subdirectory(src/utils)
+add_subdirectory_ifdef(CONFIG_DESH_DECT_PHY src/dect)
+if(NOT ENV{PROJECT_NAME})
+  zephyr_compile_definitions(
+    PROJECT_NAME=${CMAKE_PROJECT_NAME}
+  )
+endif()
+
+find_package(Git QUIET)
+if(NOT APP_VERSION AND GIT_FOUND)
+  execute_process(
+    COMMAND ${GIT_EXECUTABLE} describe --abbrev=12
```

Comment:
this will not work for incrememental builds, for example this code is not triggered in the following case:
- `cmake ...-Bbuild samples/dect/dect_phy/dect_shell`
- `ninja -C build`
- edit a source file, for example `samples/dect/dect_phy/dect_shell/src/main.c`
- commit changes
- rebuild: `ninja -C build`

`APP_VERSION` is now wrong.

See also https://github.com/zephyrproject-rtos/zephyr/issues/39503 why the principle used in this example is flawed.

Reply:
This is similar that what for example the mosh is getting from lib/modem_info for printing version: https://github.com/nrfconnect/sdk-nrf/blob/3145979b16b65468d3e988b88d97a71ab0c5c297/samples/dect/dect_phy/dect_shell/src/utils/desh_print.c#L153

In our usage, mainly for CI (that is the main source for desh/mosh images) where we are doing build from a scratch, it has done what is expected and what is really needed.

Shall I add some comment due to this or is there some better way to do this?

Reply:
> This is similar that what for example the mosh is getting from lib/modem_info for printing version:

Funny that you mention `lib/modem_info`, because back in 2021 I pointed out that this was flawed, as from earlier link I provided:
https://github.com/zephyrproject-rtos/zephyr/issues/39503#issuecomment-962010112
> But the problem is that users might copy this code and think it's a good and a trustworthy way to include current git revision into the build and not realizing that it's actually flawed.
> Just like here:
> https://github.com/nrfconnect/sdk-nrf/blob/17999765cd133f2816bdc0fe9007fac01b3d79fb/lib/modem_info/CMakeLists.txt#L12-L28

So can we avoid copying broken stuff to other places ?

Reply:
@tejlmand, how this should be done?

We do not want application or library specific version here, which I assume VERSION file would provide. We were instructed not to use application specific versioning when we initially added modem_shell into sdk-nrf because nrf version (git hash) is very relevant from the functionality perspective and not just the sample code itself. And I agreed with this principle.

We'll remove it here so that we can proceed with the PR but in general I'd like to fix modem_info and modem_shell also from this perspective.

Reply:
Add a `VERSION` file and get rid of all this cmake, e.g. https://github.com/nrfconnect/sdk-nrf/blob/main/samples/bluetooth/fast_pair/locator_tag/VERSION

Reply:
VERSION file solution is not what we want. We were specifically instructed not to use application/sample/library specific versioning when we introduced modem_shell few years ago.

Reply:
Who told you that?

Reply:
> So can we avoid copying broken stuff to other places ?

@tejlmand, OK, removed that stuff now from DeSh

Reply:
@trantanen the proper way is to fetch the git revision at build time.

To do so you define a custom command together with a custom target which then generates a header file that can be included by the source files which needs the knowledge.
The existing design of passing the setting as a compile definition in CMake also suffers from the incremental build issue described in https://github.com/zephyrproject-rtos/zephyr/issues/40167 , and should therefore be avoided.

An example on how to create such header at build time can be seen in: https://github.com/zephyrproject-rtos/zephyr/pull/42527/commits/fd265d00ab70779fee4c81382fb2f13ff0c94ab1

but I too would like to know why you're not using the common `VERSION` infrastructure, which defines a `APP_BUILD_VERSION` for the sample and is equal to: 

![image](https://github.com/user-attachments/assets/140dffa0-52d1-4da0-b909-5c75eb490856)	

Ref: https://docs.zephyrproject.org/latest/build/version/index.html#use-in-code